src/main/java/com/google/cloud/pubsublite/kafka/SingleSubscriptionConsumerImpl.java

```diff
@@ -102,30 +93,30 @@
   public void setAssignment(Set<Partition> assignment) {
     try (CloseableMonitor.Hold h = monitor.enter()) {
 
-      List<SubscriberState> unassigned =
+      List<SinglePartitionSubscriber> unassigned =
           ImmutableSet.copyOf(partitions.keySet()).stream()
               .filter(p -> !assignment.contains(p))
               .map(partitions::remove)
               .collect(Collectors.toList());
-      for (SubscriberState state : unassigned) {
-        state.subscriber.close();
-        state.committer.stopAsync().awaitTerminated();
+      for (SinglePartitionSubscriber subscriber : unassigned) {
+        subscriber.close();
       }
       assignment.stream()
           .filter(p -> !partitions.containsKey(p))
           .forEach(
               ExtractStatus.rethrowAsRuntime(
                   partition -> {
-                    SubscriberState s = new SubscriberState();
-                    s.subscriber =
-                        subscriberFactory.newPullSubscriber(
+                    Committer committer = committerFactory.newCommitter(partition);
+                    committer.startAsync().awaitRunning();
+                    SinglePartitionSubscriber subscriber =
```

Comment:
structuring nit. take an unstarted committer and use a ProxyService

Reply:
Done